scipy/special/_support_alternative_backends.py

```diff
@@ -79,6 +43,8 @@ def __xlogy(x, y, *, xp=xp):
 
 
 def _chdtr(xp, spx):
+    if spx is None:
```

Comment:
We had moved these to:
```
if f_name in _generic_implementations and spx is not None:
```
and I'm moving them back because some `generic_implementation` functions (e.g. `rel_entr`) only need `xp`, not `spx`.